### ISSUE TITLE
Hello sir, I would like to ask you about the error that is reported after starting the launch file, which shows that the drone formation is not possible.

### ISSUE THREAD

**Issue by lgl-960709** [burkap/swarm#1]
![2022-03-31 15-52-37屏幕截图](https://user-images.githubusercontent.com/82514690/161005668-5114fbdc-2fd2-4c38-b6fc-89b351b1c739.png)
![2022-03-31 15-53-00屏幕截图](https://user-images.githubusercontent.com/82514690/161005688-90357ad8-10e9-4817-9659-1cb51aaf8d94.png)
![2022-03-31 15-53-47屏幕截图](https://user-images.githubusercontent.com/82514690/161005709-61596680-b876-41f6-8452-d693ccb82519.png)
Look forward to your guidance

**Comment by burkap** [comment]
It seems that the scipy Python module is not available on your computer. Try to download the package with `pip install scipy` or `pip2 install scipy` depending on your Python version.

**Comment by lgl-960709** [comment]
Hello sir, can I read your paper on this code. Thank you very much.

**Comment by kc52** [comment]
Hello, When trying to launh the source code, the simulation starting but there is no action on gazebo. why this error occuring ?

**Comment by kc52** [comment]
could you pls share your email or any contact information ?

**Comment by burkap** [comment]
@lgl-960709 Sorry for the late response, this project is mostly based on the paper "Decentralized Formation Control of a Swarm of Quadrotor Helicopters" Also, I want to remind you that this codebase is not ready to use, it is just a proof of concept.


@kc52 I'm not sure what the problem is but it might be related to missing packages. Please try the instructions below.
> In the event that the simulation does not start, the problem may be related to Gazebo and missing packages. Therefore, run the following commands. 
```
$ sudo apt-get remove ros-melodic-gazebo* gazebo*
$ sudo sh -c 'echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list'
$ wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
$ sudo apt-get update
$ sudo apt-get install gazebo9 gazebo9-* ros-melodic-gazebo-*
$ sudo apt upgrade
```

**Comment by kc52** [comment]
THANK YOU for your valuable and fast response. On the other hand, can we realize this formation via different quadrotor for example: iris or hector ?

**Comment by burkap** [comment]
Yes, the logic behind the formation control is unrelated to the agent. You can use any quadrotor if you have the controller available for velocity or position.
